Library/Formula/llvm.rb

```diff
@@ -78,17 +101,27 @@ class Llvm < Formula
       url "http://llvm.org/git/lldb.git"
     end
 
-    resource "clang-tools-extra" do
-      url "http://llvm.org/git/clang-tools-extra.git"
+    resource "polly" do
+      url "http://llvm.org/git/polly.git"
+    end
+
+    resource "compiler-rt" do
+      url "http://llvm.org/git/compiler-rt.git"
     end
   end
 
+  keg_only :provided_by_osx
+
   option :universal
-  option "with-clang", "Build Clang support library"
+  option "with-clang", "Build the Clang compiler and support libraries"
+  option "with-clang-extra-tools", "Build extra tools for Clang"
+  option "with-libcxx", "Build the libc++ standard library"
   option "with-lld", "Build LLD linker"
   option "with-lldb", "Build LLDB debugger"
   option "with-rtti", "Build with C++ RTTI"
+  option "with-polly", "Build with the experimental Polly optimizer"
```

Comment:
You might want to put that between `with-lldb` and `with-rtti` to keep the options sorted.